streaming/rdm_atomic.c

```diff
@@ -170,63 +243,87 @@ static int run_op(void)
 	case FI_LXOR:
 	case FI_BXOR:
 	case FI_ATOMIC_WRITE:
-		for (datatype = 0; datatype < FI_DATATYPE_LAST; datatype++) {
-			ret = check_base_atomic_op(ep, op_type, datatype, count);
+		if (run_all_datatypes) {
+			for (datatype = 0; datatype < FI_DATATYPE_LAST; datatype++) {
+				ret = check_base_atomic_op(ep, op_type,
+							   datatype, count);
+				if (ret == -FI_ENOSYS || ret == -FI_EOPNOTSUPP) {
+					fprintf(stderr,
+						"Provider doesn't support %s ",
+						fi_tostr(&op_type,
+							 FI_TYPE_ATOMIC_OP));
+					fprintf(stderr,
+						"compare atomic operation on %s\n",
+						fi_tostr(&datatype,
+							 FI_TYPE_ATOMIC_TYPE));
+					continue;
+				} else if (ret) {
+					break;
+				}
+
+				ret = execute_atomic_op(base, test_name, opts,
+							op_type, datatype);
+			}
+		} else {
+			ret = check_base_atomic_op(ep, op_type,
+						   datatype, count);
 			if (ret == -FI_ENOSYS || ret == -FI_EOPNOTSUPP) {
-				fprintf(stderr, "Provider doesn't support %s ",
-					fi_tostr(&op_type, FI_TYPE_ATOMIC_OP));
-				fprintf(stderr, "base atomic operation on %s\n",
-					fi_tostr(&datatype, FI_TYPE_ATOMIC_TYPE));
-				continue;
+				fprintf(stderr,
+					"Provider doesn't support %s ",
+					fi_tostr(&op_type,
+						 FI_TYPE_ATOMIC_OP));
+				fprintf(stderr,
+					"compare atomic operation on %s\n",
```

Comment:
"base atomic operation"

Reply:
Given that there were two errors because of copy-paste, maybe we should merge the check_xx_atomic_op() and subsequent check inside of the execute_atomic_op() macro?

Reply:
@shefty I completely agree with you. That would great to have it and avoid such stupid errors.